### PR TITLE
[libcommhistory] Fix CallModel grouping by contacts. Contributes to MER#1259

### PR DESCRIPTION
**Pull request by matthewvogt**
Grouping by contact was not correctly handled where resolution was provided asynchronously.

### CODE REVIEW

File: src/callmodel.cpp [hunk 1]
```diff
@@ -61,6 +60,57 @@ using namespace CommHistory;
...
  * ************************************************************************** */
 
+class CallModelPrivate : public EventModelPrivate
+{
+    Q_OBJECT
+    Q_DECLARE_PUBLIC( CallModel )
+
+public:
+    CallModelPrivate( EventModel *model );
+
```
Comment (chriadam) [inline src/callmodel.cpp]: why all the whitespace?

Comment (matthewvogt) [inline src/callmodel.cpp]: Matching the surrounding code.

Comment (matthewvogt) [inline src/callmodel.cpp]: Actually, this is the verbatim pre-existing content of callmodel_p.h, moved into the implementation file.

**Comment by matthewvogt** [comment]
Added s test which demonstrates the effect of multiple contact matches.  In practice, the result is unpredictable - this could be fixed to be consistent in libcontacts, but I'm not sure that this would be an improvement.  Ideally, commhistory events would be associated with contacts as well as addresses, but that goes against the entire existing design.

**Comment by chriadam** [comment]
LGTM

**Pull request merged**